superset/db_engine_specs/__init__.py

```diff
@@ -36,6 +36,7 @@
 from superset.db_engine_specs.base import BaseEngineSpec
 
 engines: Dict[str, Type[BaseEngineSpec]] = {}
+_translates: Dict[str, str] = {"postgres": "postgresql"}
```

Comment:
Good idea! Let's add a property `BaseEngineSpecs.aliases: List[str] = []` (I'm open to naming suggestions, perhaps 'engine_aliases' might be even more descriptive). This way we can then override this in `PostgresEngineSpec` by defining `aliases = ["postgres"]`.